### DISCUSSION TITLE
Permanent "Skip to main content" bar on top of header at https://matplotlib.org/stable/index.html

### DISCUSSION THREAD

**Discussion by adigitoleo** [qutebrowser/qutebrowser#7583] [General] [answered]
On QB, when I go to the Matplotlib docs at https://matplotlib.org/stable/index.html, there is a bar across the top with a link titled "Skip to main content". The bar persists regardless of window size, and obscures things like the page title, github link, and momst importantly the search box. Does anyone else get this as well?

![matplotlib-3 6 3-docheader](https://user-images.githubusercontent.com/34595875/218373476-b1eadcc3-f110-4f0d-bdfa-c1979f0a5c0c.png)

```
Version info
         ______     ,,
    ,.-"`      | ,-` |
  .^           ||    |
 /    ,-*^|    ||    |
;    /    |    ||    ;-*```^*.
;   ;     |    |;,-*`         \
|   |     |  ,-*`    ,-"""\    \
|    \   ,-"`    ,-^`|     \    |
 \    `^^    ,-;|    |     ;    |
  *;     ,-*`  ||    |     /   ;;
    `^^`` |    ||    |   ,^    /
          |    ||    `^^`    ,^
          |  _,"|        _,-"
          -*`   ****"""``

qutebrowser v2.5.2
Git commit: 
Backend: QtWebEngine 5.15.12, based on Chromium 87.0.4280.144
Qt: 5.15.8

CPython: 3.10.9
PyQt: 5.15.9

sip: 6.7.6
colorama: 0.4.6
jinja2: 3.1.2
pygments: 2.14.0
yaml: 6.0
adblock: no
PyQt5.QtWebEngineWidgets: yes
PyQt5.QtWebEngine: 5.15.6
PyQt5.QtWebKitWidgets: no
pdf.js: 3.2.146 (/usr/share/pdf.js/build/pdf.js)
sqlite: 3.40.1
QtNetwork SSL: OpenSSL 3.0.8 7 Feb 2023

Style: Qt5CTProxyStyle
Platform plugin: wayland
OpenGL: AMD, 4.6 (Compatibility Profile) Mesa 22.3.4
Platform: Linux-6.1.9-arch1-2-x86_64-with-glibc2.37, 64bit
Linux distribution: Arch Linux (arch)
Frozen: False
Imported from /usr/lib/python3.10/site-packages/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt

Paths:
cache: /home/admin/.cache/qutebrowser
config: /home/admin/.config/qutebrowser
data: /home/admin/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser

Autoconfig loaded: no
Config.py: /home/admin/.config/qutebrowser/config.py has been loaded
```

**Comment by adigitoleo** [comment]
Forgot to mention: it doesn't happen on Firefox or Edge, those are the only others I've tried.

**Comment by The-Compiler** [comment, marked as answer]
I can reproduce this with QtWebEngine but not Chromium indeed - it does seem to be something fixed in newer versions of Chromium though, so nothing qutebrowser or QtWebEngine could do about it.

Bisected in Chromium via `python tools/bisect-builds.py -a linux64 -b 90.0.4430.85 -g 109.0.5414.74 --use-local-cache --verify-range -- --no-first-run --user-data-dir=/tmp --disable-seccomp-filter-sandbox https://matplotlib.org/stable/index.html` which gives me:

> You are probably looking for a change made after 1011183 (known bad), but no later than 1011196 (first known good).
>
> CHANGELOG URL:
>  https://chromium.googlesource.com/chromium/src/+log/79ab62321a00bc00b092b61f53729e491f2faaba..29e5435df96e377b6e48e2084cd7be6beb8a0a37

And it seems to be this they're using:

- [Enable CSS Individual Transform Properties (CSSIndependentTransformProperties flag) for stable. (3690616) · Gerrit Code Review](https://chromium-review.googlesource.com/c/chromium/src/+/3690616)
- [496550 - CSS Individual Transform Properties (previously known as Independent Transform Properties) - chromium](https://bugs.chromium.org/p/chromium/issues/detail?id=496550)
- [Individual Properties for CSS Transforms - Chrome Platform Status](https://chromestatus.com/feature/5705698193178624)
- [Intent to Ship: Independent/Individual Properties for CSS Transforms](https://groups.google.com/a/chromium.org/g/blink-dev/c/hu8e23c7tzg/m/2NsUPHNnBwAJ)

So that website is just using a [rather new CSS feature](https://caniuse.com/mdn-css_properties_translate) (Chromium 104, [from August](https://developer.chrome.com/blog/new-in-chrome-104/)).

If you are using the Qt 6 branch (#7202) with a new enough Qt, you can use `--qt-flag enable-blink-features=CSSIndependentTransformProperties` (or add that to the `qt.args` setting) to make it work. I'm assuming it will work out of the box with [Qt 6.5 soon](https://wiki.qt.io/Qt_6.5_Release) (on the `qt6-v2` branch as well, though).

**Reply by adigitoleo** [reply]
I see, thanks for looking into it. Glad to hear it's just a matter of waiting for the update, not something more permanent. I'll keep an eye out and upgrade my Qt when I get a chance.
